### PR TITLE
feat: Add custom root directory (#134)

### PR DESCRIPTION
**Pull request by j-tag**
As I mentioned in #134 , it would be great if we have the ability to set custom directory for blog so we can use the compiled version in another machine.

In this PR I added the functionality to set custom directory for blog.

### CODE REVIEW

**Review by roj1512 — COMMENTED** [review]
Can you add a test for that?

**Comment by J-TAG** [comment]
Yea for sure. I'll add test and update my PR

**Comment by KorigamiK** [comment]
The name `customDirectory` should be renamed to something like `contentDirectory` to be more appropriate.

**Comment by J-TAG** [comment]
> The name `customDirectory` should be renamed to something like `contentDirectory` to be more appropriate.

I'm adding the ability of setting base directory of blog in my PR. In my opinion `contentDirectory` should be the name of our property when we want to change the `posts` directory.

But in this PR the `posts` directory will still remain the same and will be like this: `customDirectory`  + `/posts`.
Correct me please if I'm wrong.

**Comment by roj1512** [comment]
How is `rootDirectory`?

**Comment by J-TAG** [comment]
`rootDirectory` seems good to me. So I rename the settings property from `customDirectory` to `rootDirectory` in my PR?

**Comment by roj1512** [comment]
Yes, and you can refer it to as “custom root directory.”

**Comment by J-TAG** [comment]
ok thank you. I'll update my PR

**Comment by J-TAG** [comment]
I've changed the name to `rootDirectory`

File: blog_test.ts [hunk 1]
```diff
@@ -338,3 +338,28 @@ Deno.test("Plaintext response", async () => {
   const body = await resp.text();
   assert(body.startsWith("It was popularised in the 1960s"));
 });
+
+Deno.test(
+  "custom root directory for blog",
```
Comment (roj1512) [inline blog_test.ts:343]:
```suggestion
  "custom root directory",
```

File: testdata/customRootDir/posts/custom.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+title: Custom post
+publish_date: 2022-03-20
+abstract: This is the first post in custom directory.
```
Comment (roj1512) [inline testdata/customRootDir/posts/custom.md:4]:
```suggestion
abstract: This is the first post with a custom root directory.
```

File: types.d.ts [hunk 3]
```diff
@@ -78,6 +78,8 @@ export interface BlogSettings {
   hostname?: string;
   /** Whether to display readtime or not */
   readtime?: boolean;
+  /** Custom root directory to looking for blog files and content */
```
Comment (roj1512) [inline types.d.ts:81]:
```suggestion
  /** The root directory of the blog content */
```

**Comment by J-TAG** [comment]
Thanks for the suggestions. I've applied them.

**Review by roj1512 — APPROVED** [review]
LGTM!

**Comment by lino-levan** [comment]
@hashrock please review.

**Review by lino-levan — APPROVED** [review]
LGTM. Thanks for the test.

**Comment by j-tag** [comment]
Thanks @hashrock for you review. My main reason for this PR is to be able to compile the blog in a machine, and host it on another. If static assets are not affected, it will not be useful. I'll try to check your point ASAP and update my PR if any changes needed.

**Comment by j-tag** [comment]
@hashrock  I've checked your point and that was totally valid. Static files are not served with the previous version of my PR. So I've fixed my PR to support static files also.
The fact that I didn't catch this in previous version was that my browser was caching images and assets so I couldn't notice deno blog is not serving them. But after a hard refresh I could see that.

So basically now if you set the `rootDirectory` option, both posts and static asset files are gonna serve in that directory. 
Also I updated the tests to check static files with a custom `rootDirectory`.

Thank you for your review!

**Review by hashrock — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**